[BUG] Sanitizer failed in arm64 test

**Describe the bug**
This is a ticket to track the status for now. I have not done multiple tests due to lack of time.
This could be an intermittent one, but we currently disabled sanitizer in https://github.com/NVIDIA/spark-rapids-jni/pull/1385.


In test build, the testing module failed init the cudf JNI test w/ sanitizer wrapper,
```
01:19:41  [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project spark-rapids-jni: There are test failures.
01:19:41  [ERROR] 
01:19:41  [ERROR] Please refer to /home/jenkins/agent/workspace/peixinl-spark-rapids-jni/target/surefire-reports for the individual test results.
01:19:41  [ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
01:19:41  [ERROR] Error occurred in starting fork, check output in log
01:19:41  [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: Error occurred in starting fork, check output in log
01:19:41  [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:624)
01:19:41  [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
01:19:41  [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
01:19:41  [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
01:19:41  [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
01:19:41  [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
01:19:41  [ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
01:19:41  [ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
01:19:41  [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
01:19:41  [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
01:19:41  [ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
01:19:41  [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
01:19:41  [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
01:19:41  [ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
01:19:41  [ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
01:19:41  [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
01:19:41  [ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
01:19:41  [ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
01:19:41  [ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:954)
01:19:41  [ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
01:19:41  [ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
01:19:41  [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
01:19:41  [ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
01:19:41  [ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
01:19:41  [ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
01:19:41  [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
01:19:41  [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
01:19:41  [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
01:19:41  [ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
01:19:41  [ERROR] -> [Help 1]
01:19:41  [ERROR] 
01:19:41  [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
01:19:41  [ERROR] Re-run Maven using the -X switch to enable full debug logging.
```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/ci/nightly-build.sh
+++ b/ci/nightly-build.sh
@@ -34,7 +34,7 @@ profiles="source-javadoc"
 if [ "${ARM64}" == "true" ]; then
   profiles="${profiles},arm64"
   USE_GDS="OFF"
-  USE_SANITIZER="OFF"
+  USE_SANITIZER="ON"
   BUILD_FAULTINJ="OFF"
 fi
 

